rag/llm/ocr_model.py

```
@@ -325,3 +327,68 @@ def parse_pdf(self, filepath: str, binary=None, callback=None, parse_method: str
+class MistralOcrModel(Base, MistralParser):
+    _FACTORY_NAME = "Mistral OCR"
+
+    def __init__(self, key: str | dict, model_name: str, **kwargs):
+        Base.__init__(self, key, model_name, **kwargs)
+        raw_config: dict = {}
+        if isinstance(key, dict):
+            raw_config = key
+        elif key:
+            try:
+                raw_config = json.loads(key)
+            except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)